fedimint-core/src/module/mod.rs

```diff
@@ -172,8 +188,9 @@ macro_rules! __api_endpoint {
                 $state: &'a Self::State,
                 $dbtx: &'a mut fedimint_core::db::ModuleDatabaseTransaction<'b, ModuleInstanceId>,
                 $param: Self::Param,
-                has_auth: bool,
+                _has_auth: bool,
             ) -> ::std::result::Result<Self::Response, $crate::module::ApiError> {
+                $(let $auth = _has_auth;)?
```

Comment:
This looks like a hack. Why not just bind the bool arg to `$auth` if present and `_` otherwise?

Reply:
Will get rid of this in a follow-up PR, since we want to put `auth` and `dbtx` into a new `ApiEndpointContext` struct.